### PR TITLE
Adopt PEP-8 for whitespace recommendations

### LINKED ISSUE #184: Adopt PEP-8 for whitespace recommendations
**Abstract**

The current style guide for whitespace (per [`CONTRIBUTING.md`](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)) recommends:

> * Use whitespaces to improve code readability (examples: after commas; before and
>  after mathematical operators (`+`/`-`/`*`/`/` except `^`), binary operators
>  (`&&`/`||`/...), and comparisons (`<`/`>`/`==`/...); before and after equality

does not follow [PEP-8 recommendations](https://peps.python.org/pep-0008/#other-recommendations), i.e.

```Python
# Correct:
i = i + 1
submitted += 1
x = x*2 - 1
hypot2 = x*x + y*y
c = (a+b) * (a-b)
```

and 

```Python
# Wrong:
i=i+1
submitted +=1
x = x * 2 - 1  # <-- current Cantera style 
hypot2 = x * x + y * y  # <-- current Cantera style 
c = (a + b) * (a - b)  # <-- current Cantera style 
```

as well as some other discrepancies.

**Motivation**

The PEP-8 recommendation seeks to improve code readability, so it should be adopted for all coding languages supported by Cantera.

**Possible Solutions**

Update style recommendations in `CONTRIBUTING.md`, and use style *going forward*. Add comment that aggressive substitutions are not encouraged.

**References**

Python PEP-8 [Whitespace in expressions and statements](https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements)

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

The current style guide for whitespace (per [`CONTRIBUTING.md`](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)) recommends:

> * Use whitespaces to improve code readability (examples: after commas; before and
>  after mathematical operators (`+`/`-`/`*`/`/` except `^`), binary operators
>  (`&&`/`||`/...), and comparisons (`<`/`>`/`==`/...); before and after equality

does not follow [PEP-8 recommendations](https://peps.python.org/pep-0008/#other-recommendations). This PR updates Cantera style guide recommendations to become consistent. 

In addition:
- Update Python documentation instructions
- Fix typo in `SConstruct`

**If applicable, fill in the issue number this pull request is fixing**

Closes Cantera/enhancements#184

**If applicable, provide an example illustrating new features this pull request is introducing**

Per PEP-8:

```Python
# Correct:
i = i + 1
submitted += 1
x = x*2 - 1
hypot2 = x*x + y*y
c = (a+b) * (a-b)
```

and 

```Python
# Wrong:
i=i+1
submitted +=1
x = x * 2 - 1  # <-- current Cantera style 
hypot2 = x * x + y * y  # <-- current Cantera style 
c = (a + b) * (a - b)  # <-- current Cantera style 
```

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -55,10 +55,13 @@
...
-* Use whitespaces to improve code readability (examples: after commas; before and
-  after mathematical operators (`+`/`-`/`*`/`/` except `^`), binary operators
-  (`&&`/`||`/...), and comparisons (`<`/`>`/`==`/...); before and after equality
-  signs `=` unless used for the assignment of a default parameter)
+* Use whitespaces to improve code readability. Examples: after commas; before and
+  after binary operators (`&&`/`||`/...), and comparisons (`<`/`>`/`==`/...); before and
+  after equality signs `=` unless used for the assignment of a default parameter. For
+  mathematical operators (`+`/`-`/`*`/`/` except `^`), consider adding whitespace around
+  the operators with the lowest priority. For additional guidance, refer to
+  [Python PEP-8](https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements),
```
Comment (speth) [inline CONTRIBUTING.md]: The wording here is pretty subtle, and I'm not sure most readers will catch it. Also, I *think* that in cases without different operator precedence like `x + y + z`, the spaces are more than just something to "consider". If you want anyone to notice this, a couple very simple examples might help, like `x*2 - 1` and `(a+b) * (a-b)` from PEP-8.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -124,13 +127,24 @@
...
 ## Python
 
+### Sphinx comments
+
+* Cantera Python documentation is based on the Python documentation generator
+  [Sphinx](https://www.sphinx-doc.org/en/master/index.html)
+* All classes, member variables, and methods should include
+  [Python docstrings](https://peps.python.org/pep-0257/#what-is-a-docstring)
+* Docstrings should use annotations compatible with
+  [automatic documentation generation from code](https://www.sphinx-doc.org/en/master/tutorial/automatic-doc-generation.html)
```
Comment (speth) [inline CONTRIBUTING.md]: Neither of these links explain how to add a Sphinx-parseable docstring to a member variable

**Review by speth — CHANGES_REQUESTED** [review]
Thanks, @ischoegl. Just a couple of comments for your consideration.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ischoegl) [reply CONTRIBUTING.md]: 👍 ... the documentation of the `autodoc` extension is actually pretty poor, but there are other resources.

**Comment by ischoegl** [comment]
Thanks, @speth. I believe this is ready to merge.

**Comment by ischoegl** [comment]
@speth ... thanks. I believe this is ready.

**Review by speth — APPROVED** [review]

**Pull request merged**